source/includes/data-modeling/field-types.rb

```diff
@@ -0,0 +1,251 @@
+# start-define-fields
+class Person
+  include Mongoid::Document
+  field :name, type: String
+  field :date_of_birth, type: Date
+  field :weight, type: Float
+end
+# end-define-fields
+
+# start-define-untyped
+class Person
+  include Mongoid::Document
+
+  field :name, type: String
+  field :preferences
+end
+# end-define-untyped
+
+# start-untyped
+product = Product.new(properties: "color=white,size=large")
+# properties field saved as String: "color=white,size=large"
+
+product = Product.new(properties: {color: "white", size: "large"})
+# properties field saved as Object: {:color=>"white", :size=>"large"}
+# end-untyped
+
+# start-stringified-symbol
+class Post
+  include Mongoid::Document
+  
+  field :status, type: StringifiedSymbol
+end
+  
+# Save status as a symbol
+post = Post.new(status: :hello)
+# status is stored as "hello" on the database, but returned as a Symbol
+post.status
+# Outputs: :hello
+
+# Save status as a string
+post = Post.new(status: "hello")
+# status is stored as "hello" in the database, but returned as a Symbol
+post.status
+# Outputs: :hello
+# end-stringified-symbol
+
+# start-hash
+class Person
+  include Mongoid::Document
+  field :first_name
+  field :url, type: Hash
+  
+  def set_vals
+    self.first_name = 'Daniel'
+    self.url = {'home_page' => 'http://www.homepage.com'}
+    save
+  end
+end
+# end-hash
+
+# start-time
+class Voter
+  include Mongoid::Document
+  
+  field :registered_at, type: Time
+end
+  
+Voter.new(registered_at: Date.today)
+# end-time
+
+# start-datetime
+class Ticket
```

Comment:
S: consider changing to be more intuitive

Reply:
Updated the fields